03-rnaseq/ortholog-mapping_rnaseq_01_ensembl.Rmd

```diff
@@ -0,0 +1,369 @@
+
+In this analysis, we will be using `hcop` to map between mouse and human identifiers [@Turner2018].
+
+We'll need to use the `devtools` package to install `hcop` from GitHub [@Wickham2020].
+
+```{r}
+if (!("devtools" %in% installed.packages())) {
+  install.packages("devtools")
+}
+```
+
+Now we can install `hcop` from GitHub.
+You can control what version of the package is installed by using the `ref` argument of `devtools::install_github()`.
+Here, we'll use the most recent commit at the time that we are putting together this example.
+
+```{r}
+devtools::install_github("stephenturner/hcop", ref = "0985fddc91a6ef2308f4800958dfd11c25fe6a98")
+```
+
+Attach the packages we need for this analysis.
+
+```{r}
+# Attach the library
+library(hcop)
+
+# We will need this so we can use the pipe: %>%
+library(magrittr)
+```
+
+## Import and set up data
+
+Data downloaded from refine.bio include a metadata tab separated values (TSV) file and a data TSV file. 
+This chunk of code will read the both TSV files and add them as data frames to your environment.
+
+```{r}
+# Read in metadata TSV file
+metadata <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your metadata file
+  "metadata_SRP070849.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data TSV file
+df <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your data file
+  "SRP070849.tsv" # Replace with the name of your data file
+)) %>%
+  # Tuck away the gene ID column as rownames
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(metadata$refinebio_accession_code)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$refinebio_accession_code)
+
+# Bring back the "Gene" column in preparation for mapping
+df <- df %>%
+  tibble::rownames_to_column("Gene")
+```
+
+## Mapping mouse Ensembl gene IDs to human symbols
+
+refine.bio data uses Ensembl gene identifiers, which will be in the first column.
+
+```{r}
+# Let's take a look at the first 6 items in the `Gene` column of `df`
+head(df$Gene)
+```
+
+Ensembl gene identifiers have different species-specific prefixes. 
+In mouse, Ensembl gene identifiers begin with `ENSMUSG` (in human, `ENSG`, etc.).
+
+Now let's do the conversion!
+We'll use the `mouse` data from `hcop`; this is what `mouse` looks like.
+
+```{r}
+mouse
+```
+
+We're interested in the `human_symbol` and `mouse_ensembl` columns specifically.
+
+```{r}
+human_mouse_key <- mouse %>%
+  # we'll want to subset mouse to only the columns we're interested in
+  dplyr::select(mouse_ensembl, human_symbol)
```

Comment:
Do you need to do the same `distinct()` steps you did in microarray? My guess is you do, but I haven't ran this myself yet.